File: c++/4mod2.cpp
#include <iostream>
#include <vector>
using namespace std;

int countDigits(const vector<int>& nums){ // Функция для подсчета цифр в числе
    int count = 0;
    for (int i = 0; i < nums.size(); i++){
        int num = abs(nums[i]); 
        int digits = 0;
        if (num == 0) digits = 1;
        else{
            while (num != 0){
                num /= 10;
                digits += 1;
            }
        }
        if (digits % 2 != 0) count += 1; // Если нечетно, то прибавляем к счетчику
    }
    return count;
}


int main(){
    int size;
    cout << "Введите размер массива" << endl;
    cin >> size;
    vector<int> numbers(size);
    for (int i = 0; i < size; i++){
        cout << i + 1 << " Элемент массива: ";
        cin >> numbers[i];
    }
    cout << countDigits(numbers) << " Чисел(а) содержат нечетное количество цифр" << endl;
    return 0;
}

File: python/4mod2.py
def countDigits(nums):
    count = 0
    for num in nums:  # идем по массиву nums (элементы num в списке nums)
        num = abs(num)
        digits = 0
        if num == 0:
            digits = 1
        else:
            while num != 0:
                num //= 10  # Используем целочисленное деление
                digits += 1
            if digits % 2 != 0:
                count += 1
    return count

def main():
    size = int(input("Введите размер массива: "))  
    nums = []  # Список для чисел
    for i in range(size):
        num = int(input(f"Число {i + 1} = "))  # используем f-строки для вывода
        nums.append(num)  # добавляем число в список
    print(f"Нечетное кол-во цифр содержат {countDigits(nums)} число(а)")  # передаем список в функцию

if __name__ == "__main__":
    main()